indico/modules/categories/forms.py

```diff
@@ -119,57 +142,103 @@ def __init__(self, *args, category, **kwargs):
-                    _('The credentials could not be loaded.')
+        if config.ENABLE_GOOGLE_WALLET:
+            enabled = self.google_wallet_mode.data == InheritableConfigMode.enabled
+            credentials = self.google_wallet_credentials.data
+            issuer_id = self.google_wallet_issuer_id.data
+
+            if (
+                credentials and
+                enabled and (
+                    self.category.google_wallet_settings.get('google_wallet_credentials') != credentials or
+                    self.category.google_wallet_settings.get('google_wallet_issuer_id') != issuer_id
                 )
-                return False
-            elif res == GoogleCredentialValidationResult.refused:
-                self.google_wallet_credentials.errors.append(
-                    _('The credentials are invalid or have no access to the Google Wallet API.')
-                )
-                return False
-            elif res == GoogleCredentialValidationResult.failed:
-                self.google_wallet_credentials.errors.append(
-                    _('There was an error validating your credentials. Contact your Indico administrator for details.')
-                )
-                return False
-            elif res == GoogleCredentialValidationResult.bad_issuer:
-                self.google_wallet_issuer_id.errors.append(
-                    _('The Issuer ID is invalid or not linked to your credentials.')
-                )
-                return False
+            ):
+                res = GoogleWalletManager.verify_credentials(credentials, issuer_id)
+                if res == GoogleCredentialValidationResult.invalid:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials could not be loaded.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.refused:
+                    self.google_wallet_credentials.errors.append(
+                        _('The credentials are invalid or have no access to the Google Wallet API.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.failed:
+                    self.google_wallet_credentials.errors.append(
+                        _('There was an error validating your credentials. Contact your Indico administrator for '
+                          'details.')
+                    )
+                    return False
+                elif res == GoogleCredentialValidationResult.bad_issuer:
+                    self.google_wallet_issuer_id.errors.append(
+                        _('The Issuer ID is invalid or not linked to your credentials.')
+                    )
+                    return False
 
         return form_valid
 
+    def validate_apple_pass_certificate(self, field):
+        try:
+            x509.load_pem_x509_certificate(field.data.encode('UTF-8'))
```

Comment:
Indico requires an UTF-8 environment - throughout the codebase we use `.encode()` and `.decode()` w/o explicitly specifying the encoding. So you can do the same :)

(this applies to any other place in the PR as well)

Reply:
Done.